### PR TITLE
Refactor init process

### PR DESCRIPTION
**Pull request by jklepatch**
https://github.com/lendingblock/loan-contracts/issues/13

### CODE REVIEW

**Comment by jklepatch** [comment]
By revising down the complexity of the test transaction and by regrouping initiation functions into the constructor, total gas cost is `4110965` (~4M) gas, down from 6.6M before.

File: migrations/3_loans.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  '0x7bb540a8208433bafd9e33b6cec4efc8fb482502bef5fe48542f3f6045e7e6af',
+  '0xa709fd3aa96d9faf770e44a5aef2f4808a6fe3a5ddf546568f36ad3a3873f31d',
+  '0xaad60a3265e1c3c0dff4ef3474d6c608ca5f7ec61bd7dcbc5a992ad057630691',
+  '0x425443',
+  '0x455448',
+];
+
+let interestPaymentTimes = [];
+let interestAmounts = [];
+for (let i = 0; i < 3; i++) {
```
Comment (QX-Lee) [inline migrations/3_loans.js]: we can't make the comparison when this is using much fewer interest and lenders

Comment (jklepatch) [reply migrations/3_loans.js]: True, but as discussed before your initial test was with quite a high number of interest and lenders. I wanted to see how it performs with more realistic figures. I believe most of the market will be for short tenors.

**Comment by jklepatch** [comment]
After changing test data to match @Qx initial test, and enabling solc optimization, new gas cost is `5045375` (5M)

File: migrations/3_loans.js (hunk 1 above)
Comment (jklepatch) [reply migrations/3_loans.js]: I have updated the figures I have used to match yours

**Pull request closed without merging**